ui/desktop/src/App.tsx

```diff
@@ -283,6 +283,12 @@ export default function App() {
   const { modalState, modalConfig, dismissModal, confirmInstall } =
     useExtensionInstallModal(addExtension);
 
+  useEffect(() => {
+    (async () => {
+      await loadCurrentChat({ setAgentWaitingMessage });
+    })();
+  }, [loadCurrentChat, setAgentWaitingMessage]);
+
```

Comment:
we have a lot of `useEffect`s going on here, I worry about race conditions here. I wonder if this one should be added to the setView one at the top, so at least we know there is a setView? also now seeing we call reactReady back even though we are not here maybe? /cc @zanesq could that explain some of the refresh crash weirdness?